pkg/cloud/vsphere/services/loadbalancer/aws/loadbalancers.go

```diff
@@ -0,0 +1,315 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"github.com/aws/aws-sdk-go/service/elbv2/elbv2iface"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-vsphere/api/v1alpha2"
+	"sigs.k8s.io/cluster-api-provider-vsphere/api/v1alpha2/cloud"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/services"
+)
+
+var (
+	sessionCache sync.Map
+)
+
+func sessionForRegion(region string) (*session.Session, error) {
+	s, ok := sessionCache.Load(region)
+	if ok {
+		return s.(*session.Session), nil
+	}
+
+	ns, err := session.NewSession(aws.NewConfig().WithRegion(region))
+	if err != nil {
+		return nil, err
+	}
+
+	sessionCache.Store(region, ns)
+	return ns, nil
+}
+
+// Service TODO
+type Service struct {
+	ELB elbv2iface.ELBV2API
+}
+
+// NewProvider TODO
+func NewProvider(config *cloud.LoadBalancerConfig) (services.LoadBalancerService, error) {
+	region := config.AwsProvider.Region
+	session, err := sessionForRegion(region)
+	if err != nil {
+		return nil, err
+	}
+	return &Service{
+		ELB: elbv2.New(session),
+	}, nil
+}
+
+func generateELBName(clusterName string) string {
+	return fmt.Sprintf("%s-endpoint", clusterName)
+}
+
+// Reconcile reconciles loadbalancing resources in aws
```

Comment:
Hi @yastij,

I suggest rewording this as:

```
// Reconcile ensures the AWS load balancer for a vSphere cluster is synchronized with the intended state.
```

I would also consider renaming this function simply to `New` since it's in a package dedicated to this service.